tests/systemd_testsuite/runner.pm

```diff
@@ -0,0 +1,73 @@
+# SUSE's openQA tests
+#
+# Copyright 2019-2020 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Run systemd upstream test cases
+# Maintainer: Martin Loviska <mloviska@suse.com>
+
+use Mojo::Base qw(systemd_testsuite_test);
+use testapi;
+
+my $test_hash;
+
+sub build_cmd {
+    my ($target, $args) = @_;
+    my @opts = ();
+    my $test = $args->{test};
+    my $dir = $args->{dir};
+
+    foreach my $k (keys %{$args->{make_opts}}) {
+        if ($args->{make_opts}->{$k}) {
+            push @opts, "$k=$args->{make_opts}->{$k}";
+        }
+    }
+
+    return sprintf('%s make -C %s%s %s', join(" ", @opts), ${dir}, ${test}, $target);
+}
+
+sub run {
+    my ($self, $args) = @_;
+    my $timeout = 900;
+    my $marker = " >>> $args->{test} has finished <<<";
+    my $logs = qr[\/var\/tmp\/systemd-test.(\w+)\/];
+
+    $self->select_serial_terminal();
+
+    assert_script_run(build_cmd('clean', $args));
+    enter_cmd(build_cmd('setup', $args));
+
+    my $out = wait_serial(qr/make: Leaving directory.*/, 240, 0, record_output => 1);
+    if ($out =~ $logs) {
+        $test_hash = $1;
+        record_info("$test_hash", 'Test logs: /var/tmp/systemd-test.' . $test_hash);
+    } else {
+        bmwqemu::diag 'Cannot find the location for logs';
+    }
+
+    my $texec = join('', '(', build_cmd('run', $args), qq/; echo "$marker \[\$?\]")/);
+    enter_cmd("$texec | tee /dev/$serialdev");
+
+    my $test_log = wait_serial(qr/$marker\s+\[\d+\]/, $timeout, 0, record_output => 1);
+
+    if (defined($test_log) && $test_log =~ qr/$marker\s+\[(\d+)\]$/) {
+        if ($1 != 0) {
+            bmwqemu::diag "$args->{test} has failed with RC => $1!";
+            $self->{result} = 'fail';
+        }
+    } else {
+        bmwqemu::diag "$args->{test} has timed out!";
+        $self->{result} = 'fail';
+    }
+}
+
+sub post_fail_hook {
+    my $lpath = sprintf('/var/tmp/systemd-test.%s/system.journal', $test_hash);
+
+    select_console('log-console');
```

Comment:
Remember to call `super` here for for the rest of the post fail hooks

Reply:
I am not entirely sure that we want many logs from the host, as the SUT runs in a nspawn container.